### ISSUE TITLE
eval should not jump when shifting between batch size and gradient accumulation

### ISSUE THREAD

**Issue by StableLlama** [bghira/SimpleTuner#2572]
It might be necessary to change the batch size and the amount of gradient accumulation (and possibly the amount of GPUs used) during a training run. E.g. when the data set is dynamically modified with `start_epoch` to add high resolution images and this causes an OOM, so it is necessary to reduce the batch size.
Or when training and the machine is breaking and it must be continued on a different machine with other GPUs (count and VRAM available).

Currently, when that is happening, the numbers of the eval are changing as well, creating jumps in the graphs and making it harder to interpret the shape of the curve, e.g. here batch size = 4, gradient accumulation = 1 was changed to  batch size = 2, gradient accumulation = 2 to fix an OOM training stop:

<img width="1862" height="613" alt="Image" src="https://github.com/user-attachments/assets/e0de6916-e748-4560-a940-1007ea5a99a2" />

It would be better when the eval value would only depend on the state of the training and not on any hyper parameters.

**Comment by bghira** [comment]
in context, these are not very meaningful jumps. they are at 2.6 and 5.1% difference - the scale only looks large of how the Y-axis is displayed. on a larger scale, the lines will be right next to each other. the reason they change is because the samples' order changes, so far as i can tell. it is always using a batch size of 1 for the dataset, but they're randomly sampled, and the way this occurs depends on the random states. i spent far too long trying to track it down.
